README.md

```diff
@@ -96,6 +96,114 @@ public class PetResource {
 
 See the [integration-tests](integration-tests) module for more information of how to use this extension. Please be advised that the extension is on experimental, early development stage.
 
+## Circuit Breaker
+
+You can add [CircuitBreaker annotation from MicroProfile Fault Tolerance](https://microprofile.io/project/eclipse/microprofile-fault-tolerance/spec/src/main/asciidoc/circuitbreaker.asciidoc)
+to your generated classes by defining the desired configuration in `application.properties`.
+
+Let's say you have the following OpenAPI definition:
+````json
+{
+  "openapi": "3.0.3",
+  "info": {
+    "title": "Simple API",
+    "version": "1.0.0-SNAPSHOT"
+  },
+  "paths": {
+    "/hello": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "text/plain": {
+                "schema": {
+                  "type": "string"
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+
+    "/bye": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "text/plain": {
+                "schema": {
+                  "type": "string"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+````
+
+And you want to configure Circuit Breaker for the `/bye` endpoint, you can do it in the following way:
+
+Add the [SmallRye Fault Tolerance extension](https://quarkus.io/guides/smallrye-fault-tolerance) to your project's `pom.xml` file:
+
+````xml
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-smallrye-fault-tolerance</artifactId>
+</dependency>
+````
+
+Assuming your Open API spec file is in `src/main/openapi/simple-openapi.json`, add the following configuration to your `application.properties` file:
+
+````properties
+quarkus.openapi-generator.spec."simple-openapi.json".base-package=org.acme.openapi.simple
+
+# Enables the CircuitBreaker extension
+quarkus.openapi-generator.CircuitBreaker.enabled=true
+
+# Defines the configuration for the GET method to the /bye endpoint
+quarkus.openapi-generator.spec."simple-openapi.json"/byeGet/CircuitBreaker/failOn = java.lang.IllegalArgumentException,java.lang.NullPointerException
+quarkus.openapi-generator.spec."simple-openapi.json"/byeGet/CircuitBreaker/skipOn = java.lang.NumberFormatException
+quarkus.openapi-generator.spec."simple-openapi.json"/byeGet/CircuitBreaker/delay = 33
+quarkus.openapi-generator.spec."simple-openapi.json"/byeGet/CircuitBreaker/delayUnit = MILLIS
+quarkus.openapi-generator.spec."simple-openapi.json"/byeGet/CircuitBreaker/requestVolumeThreshold = 42
+quarkus.openapi-generator.spec."simple-openapi.json"/byeGet/CircuitBreaker/failureRatio = 3.14
+quarkus.openapi-generator.spec."simple-openapi.json"/byeGet/CircuitBreaker/successThreshold = 22
```

Comment:
Hi! Instead of replicating the properties from the Circuit Breaker extension, what do you think about reusing them?

See: https://quarkus.io/guides/smallrye-fault-tolerance#runtime-configuration

In this case, the properties would be:

```properties
org.acme.openapi.simple.DefaultApi/byeGet/CircuitBreaker/failOn = java.lang.IllegalArgumentException,java.lang.NullPointerException
org.acme.openapi.simple.DefaultApi/byeGet/CircuitBreaker/skipOn = java.lang.NumberFormatException
org.acme.openapi.simple.DefaultApi/byeGet/CircuitBreaker/delay = 33
org.acme.openapi.simple.DefaultApi/byeGet/CircuitBreaker/delayUnit = MILLIS
org.acme.openapi.simple.DefaultApi/byeGet/CircuitBreaker/requestVolumeThreshold = 42
org.acme.openapi.simple.DefaultApi/byeGet/CircuitBreaker/failureRatio = 3.14
org.acme.openapi.simple.DefaultApi/byeGet/CircuitBreaker/successThreshold = 22
```

This way we delegate to the proper extension to do the right configuration.

The generated code then will have only the annotations for `CircuitBreaker`s, `Retry`, etc:

```java
@Path("")
@RegisterRestClient
public interface DefaultApi {
    @GET
    @Path("/bye")
    @Produces({"text/plain"})
    @CircuitBreaker() //simple annotation, the properties are fine tuned via properties file
    public String byeGet();
    @GET
    @Path("/hello")
    @Produces({"text/plain"})
    public String helloGet();
}
```

We just need to explain how the class fqdn is generated (`base-package` + tag + Api prefix), so the user would know how to set the properties. Alternatively, they can generate the code first and add the properties later having the final class generated. :)

Reply:
Actually, it was my initial idea and I thought the user having to know the FQCN would be a problem.
But on second thought, it is a good price to pay to delegate almost everything to SmallRye.
I switched the PR to draft until I do this change.